pkg/test/framework/components/istio/ingress.go

```diff
@@ -86,8 +87,14 @@ type ingressImpl struct {
 func (c *ingressImpl) getAddressInner(cluster resource.Cluster, ns string, port int) (interface{}, bool, error) {
```

Comment:
The API here is still being changed (it no longer only returns the ingress gateway address). It feels like it might be better to do something like this?
```go
func (c *ingressImpl) getAddressInner(cluster resource.Cluster, ns string, port int) (interface{}, bool, error) {
	return getRemoteServiceAddress(cluster, ns, istioLabel, ingressServiceName, port)
}
```

```go
func (i *operatorComponent) RemoteDiscoveryAddressFor(cluster resource.Cluster) (net.TCPAddr, error) {
	if ... {
		return getRemoteServiceAddress(cluster, ns, "pilot", "istiod", port)
	}
	...
}
```

Reply:
Thanks for the detailed suggestion. fixed.